src/DynamoCore/Library/FunctionDescriptor.cs

```diff
@@ -318,49 +318,59 @@ public IEnumerable<Tuple<string, string>> InputParameters
             private set;
         }
 
+        private string category;
         /// <summary>
         ///     The category of this function.
         /// </summary>
         public string Category
         {
             get
             {
+                if (category != null)
+                {
+                    return category;
+                }
                 var categoryBuf = new StringBuilder();
                 categoryBuf.Append(GetRootCategory());
 
                 //if this is not BuiltIn function search NodeCategoryAttribute for it
                 if (ClassName != null)
                 {
-                    //get function assembly
-                    var asm = AppDomain.CurrentDomain.GetAssemblies()
-                        .Where(x => x.GetName().Name == Path.GetFileNameWithoutExtension(Assembly))
-                        .ToArray();
-
-                    if (asm.Any() && asm.First().GetType(ClassName) != null)
+                    try
```

Comment:
would it help to store a reference to the actual loaded Assembly  instead of the Path ? We already have access to the loaded assembly in the class constructor.

Reply:
hmm yes probably, will give some thought to the pros/cons.